idaes/core/util/tests/test_model_diagnostics.py

```diff
@@ -0,0 +1,249 @@
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, University of Notre Dame, et al. 
+# All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+This module contains model diagnostic utility functions for use in IDAES (Pyomo) models.
+"""
+
+import pytest
+
+# Need to update
+import pyomo.environ as pyo
+                           
+# TODO: Add pyomo.dae test case
+'''
+from pyomo.environ import TransformationFactory
+from pyomo.dae import ContinuousSet, DerivativeVar
+'''
+
+# Need to update
+from idaes.core.util.model_diagnostics import *
+
+
+# Author: Alex Dowling
+
+# This was from 
+# @pytest.fixture()
+def problem1():
+    m = pyo.ConcreteModel()
+
+    m.I = pyo.Set(initialize=[i for i in range(5)])
+
+    m.x = pyo.Var(m.I,bounds=(-10,10),initialize=1.0)
+
+    m.con1 = pyo.Constraint(expr=m.x[0] + m.x[1] - m.x[3] >= 10)
+    m.con2 = pyo.Constraint(expr=m.x[0]*m.x[3] + m.x[1] >= 0)
+    m.con3 = pyo.Constraint(expr=m.x[4]*m.x[3] + m.x[0]*m.x[3] - m.x[4] == 0)
+
+    m.obj = pyo.Objective(expr=sum(m.x[i]**2 for i in m.I))
+
+    return m
+
+def example2(with_degenerate_constraint=True):
+    ''' Create the Pyomo model for Example 2
+    
+    Arguments:
+        with_degenerate_constraint: Boolean, if True, include the redundant linear constraint
+    
+    Returns:
+        m2: Pyomo model
+    '''
+    
+    m2 = pyo.ConcreteModel()
+
+    m2.I = pyo.Set(initialize=[i for i in range(1,4)])
+
+    m2.x = pyo.Var(m2.I,bounds=(0,5),initialize=1.0)
+
+    m2.con1 = pyo.Constraint(expr=m2.x[1] + m2.x[2] >= 1)
+    m2.con2 = pyo.Constraint(expr=m2.x[1] + m2.x[2] + m2.x[3] == 1)
+    m2.con3 = pyo.Constraint(expr=m2.x[2] - 2*m2.x[3] <= 1)
+    m2.con4 = pyo.Constraint(expr=m2.x[1] + m2.x[3] >= 1)
+    
+    if with_degenerate_constraint:
+        m2.con5 = pyo.Constraint(expr=m2.x[1] + m2.x[2] + m2.x[3] == 1)
+
+    m2.obj = pyo.Objective(expr=sum(m2.x[i] for i in m2.I))
+    
+    return m2
+
+def extract_constraint_names(cs):
+    ''' Get constraint names from ComponentSet
+    
+    Arguments:
+        cs: ComponentSet object
+        
+    Return:
+        constraint_names: list of constraint names (strings)
+            
+    '''
+    
+    constraint_names = []
+    for i in cs:
+        constraint_names.append(i.name)
+    return constraint_names
+
+# Problem 1
+@pytest.mark.unit
+def test_problem1():
+    # Create test problem
+    m = problem1()
+    
+    # Specify Ipopt as the solver
+    opt = pyo.SolverFactory('ipopt')
```

Comment:
Could you add skip conditions to these tests if ipopt is not available? See here for an example: https://github.com/IDAES/idaes-pse/blob/main/idaes/tests/test_logger.py#L95

Reply:
Personally, I have been pushing to drop the `skipifs`. Seeing as we have IPOPT easily available as part of get_extensions, I see no reason why we should not expect anyone running the tests to have IPOPT available.

Reply:
I think we should keep the skipifs until we get the Mac ipopt binary added to our extensions

Reply:
I added it, but agree we should remove it once we have macOS binaries.